src/helpers/DataRequirementHelpers.ts

```diff
@@ -391,3 +391,57 @@ export function codeLookup(dataType: string, attribute: string): string | null {
   }
   return null;
 }
+
+/**
+ * Extracts the measurement period information from either the options or effective period (in that order depending on presence)
+ * and populates a parameters object including the extracted info to be passed into the parseQueryInfo function
+ */
+export function extractDataRequirementsMeasurementPeriod(options: CalculationOptions, effectivePeriod?: fhir4.Period) {
+  if (!hasMeasurementPeriodInfo(options, effectivePeriod)) {
+    return {};
+  }
+  const parameters: Record<string, Interval> = {};
+
+  if (options.measurementPeriodStart || options.measurementPeriodEnd) {
+    parameters['Measurement Period'] = createIntervalFromEndpoints(
+      options.measurementPeriodStart,
+      options.measurementPeriodEnd
+    );
+  } else {
+    parameters['Measurement Period'] = createIntervalFromEndpoints(effectivePeriod?.start, effectivePeriod?.end);
+  }
+  return parameters;
+}
+
+/**
+ * Creates a cql-execution interval from start to end. If either start or end is not present,
+ * creates an interval with duration exactly one year using the present endpoint
+ */
+export function createIntervalFromEndpoints(start?: string, end?: string) {
```

Comment:
Out of curiosity, before our default was from 2019-01-01 to 2019-12-31 but now we have intervals from the first of one year to the first of the next (2019-01-01 to 2020-01-01 for example). Was there a reason for this change?

Reply:
In all honesty just because `setYear(getYear() + 1)` was easier to type than `setYear(getYear() + 1).setDays(getDays() - 1)`. I can't imagine any consequences for adding a day to our default measurement period, but happy to change it if there's something I'm missing :)

Reply:
I agree I don't see anything wrong with it I was just making sure that was intentional!